configs/flat/typescript.mjs

```diff
@@ -1,38 +1,34 @@
-// @ts-check
-import typescriptEslint from '@typescript-eslint/eslint-plugin';
-import typescriptEslintParser from '@typescript-eslint/parser';
+import tseslint from 'typescript-eslint';
 import typescriptRuleSetBase from '../../rules/typescript.js';
 
 const typescriptRuleSet = {
   plugins: {
-    '@typescript-eslint': typescriptEslint,
-  },
-  rules: {
-    ...typescriptEslint.configs['eslint-recommended'].overrides?.[0].rules,
-    ...typescriptEslint.configs['strict-type-checked'].rules,
-    ...typescriptEslint.configs['stylistic-type-checked'].rules,
-    ...typescriptRuleSetBase.rules,
+    '@typescript-eslint': tseslint.plugin,
   },
+  extends: [
+    ...tseslint.configs.strictTypeChecked,
+    ...tseslint.configs.stylisticTypeChecked,
+  ],
+  rules: typescriptRuleSetBase.rules,
 };
 
-export default [
-  {
-    files: ['**/*.@(ts|tsx|cts|mts)'],
+export default tseslint.config({
+  files: ['**/*.@(ts|tsx|cts|mts)'],
 
-    languageOptions: {
-      parser: typescriptEslintParser,
-      parserOptions: {
-        project: true,
-      },
+  languageOptions: {
+    parser: tseslint.parser,
+    parserOptions: {
+      projectService: true,
+      tsconfigRootDir: process.env.TSCONFIG_ROOT_DIR ?? import.meta.dirname,
```

Comment:
This property is necessary to automatically detect the nearest `tsconfig.json`.

However, it's not desirable to reference `import.meta.dirname` when running snapshot test, so a dummy path is assigned to `process.env.TSCONFIG_ROOT_DIR` only during snapshot test to prevent `import.meta.dirname` from being referenced.

see: [@typescript-eslint/parser | typescript-eslint](https://typescript-eslint.io/packages/parser/#tsconfigrootdir)